src/inmanta/compiler/__init__.py

```diff
@@ -337,22 +338,28 @@ def call_finalizers(cls, should_log: bool = False) -> None:
         by default this function will raise exceptions caused by errors in the finalizer functions
         if 'should_log' is set to True the exceptions will not be raised but logged instead.
         """
-        try:
-            for fnc in cls.__finalizers:
+        excns: list[CompilerException] = []
+        for fnc in cls.__finalizers:
+            try:
                 fnc()
-        except Exception as e:
+            except Exception as e:
+                excns.append(CompilerException("Finalizer failed: " + str(e)))
+        if excns:
             if should_log:
-                LOGGER.error(f"Finalizers failed: {e}")
+                msg: list[str] = []
+                for exception in excns:
+                    msg.append(exception.msg + "\n")
+                LOGGER.error("".join(msg).strip())
             else:
-                raise
+                raise MultiException(excns)
```

Comment:
I think the `MultiException` is really only meant for 2+ execptions so I'm afraid we'll need another branch here:
```suggestion
                raise excns[0] if len(excns) == 1 else MultiException(excns)
```

Reply:
shouldn't we change the MultiException than? I mean we likely need to add this condition every time we want to raise a MultiExcpetion

Reply:
I don't think it would really break anything, but its definition is that it's a wrapper around a multitude of exceptions. But I'm not against adding a guard. @wouterdb wdyt?

Reply:
I don't understand what is exactly suggested here?

Reply:
`inmanta.ast.MultiException`, should it ever be used with a set of 1 child exceptions? If not, do we add a guard to its constructor that prevents it.